src/parser/jobs/gnb/modules/SonicBreak.tsx

```diff
@@ -0,0 +1,175 @@
+import DISPLAY_ORDER from 'parser/jobs/gnb/modules/DISPLAY_ORDER'
+import React, {Fragment} from 'react'
+import {Button, Message, Table} from 'semantic-ui-react'
+
+const MAX_TICKS = 10  // Sonic Break is 30s
+
+class SonicBreakApplication {
+	start: number
+	end?: number
+	totalTicks: number
+	isMissingTicks: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+		this.totalTicks = 0
+	}
+}
+
+export class SonicBreak extends Analyser {
+	static override handle = 'Sonic Break'
+	static override title = t('gnb.sonic-break.title')`Sonic Break`
+	static override displayOrder = DISPLAY_ORDER.SONIC_BREAK
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+
+	private SonicBreakApplications: SonicBreakApplication[] = []
+
+	private get lastSonicBreakApplication(): SonicBreakApplication | undefined {
+		return _.last(this.SonicBreakApplications)
+	}
+
+	override initialise() {
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.SONIC_BREAK.id), this.onDotApply)
+		this.addEventHook(playerFilter.type('damage'), this.onDotDamage)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.SONIC_BREAK.id), this.onDotRemove)
+		this.addEventHook('complete', this.onComplete)
+
+	}
+
+	private onDotApply(event: Events['statusApply']) {
+		const lastSonicBreak = this.lastSonicBreakApplication
+
+		//If below is true, we broke it and the last window was not closed properly.
+		if (lastSonicBreak != null && lastSonicBreak.end == null) {
+			lastSonicBreak.end = (lastSonicBreak.start + this.data.statuses.SONIC_BREAK.duration)
+		}
+
+		this.breakMaker(event)
+
+	}
+
+	private onDotDamage(event: Events['damage']) {
+		if (event.cause.type === 'status') {
+			if (event.targets.find(e => e.amount !== 0) && event.cause.status === this.data.statuses.SONIC_BREAK.id) { //Dot did damage, increment.
+				this.breakUpdater(event)
+			}
+		}
+	}
+
+	private onDotRemove(event: Events['statusRemove']) {
+		this.breakEnder(event)
+	}
+
+	private breakMaker(event: Event) : void {
+		const sonicBreakWindow = new SonicBreakApplication(event.timestamp)
+		this.SonicBreakApplications.push(sonicBreakWindow)
+	}
+
+	private breakUpdater(event: Event) : void {
+		const lastSonicBreak = this.lastSonicBreakApplication
+
+		//We missed the break maker somehow and now we need to make it.
+		if (lastSonicBreak == null || lastSonicBreak.end != null) {
+			this.breakMaker(event)
+		}
+
+		if (lastSonicBreak != null && lastSonicBreak.end == null) {
+			lastSonicBreak.totalTicks++
+		}
+	}
+
+	private breakEnder(event: Event) : void {
+		const lastSonicBreak = this.lastSonicBreakApplication
+
+		if (lastSonicBreak != null && lastSonicBreak.end == null) {
+
+			lastSonicBreak.end = event.timestamp
+
+			if (lastSonicBreak.totalTicks < MAX_TICKS) {
```

Comment:
_Nit:_ This can be simplified to `lastSonicBreak.isMissingTicks = lastSonicBreak.totalTicks < MAX_TICKS` to avoid the extra level of nesting.